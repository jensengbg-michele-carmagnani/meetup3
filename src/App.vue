<template>
  <div id="app">
    <div id="nav">
      <router-link to="/">Casa</router-link> |
      <router-link to="/login">About</router-link>
    </div>
    <router-view  />
  </div>
</template>
<script>
export default {
  beforeMount() {
    this.$store.dispatch("fetchEvents");
  },
  
};
</script>

<style lang="scss">
@import "./scss/main";
</style>
