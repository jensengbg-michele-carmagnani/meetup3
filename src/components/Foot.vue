<template>
  <section id="foot">
    <h1>footer</h1>
    <ul>
      <li></li>
      <li></li>
      <li></li>
      <li></li>
      <li></li>
      <li></li>
    </ul>
  </section>
</template>

<script>
export default {
  name: "Foot",
};
</script>

<style></style>
