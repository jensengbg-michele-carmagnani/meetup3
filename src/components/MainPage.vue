<template>
  <section id="main-page">
    <h2 class="title">Eventpage</h2>
    <section class="search-section">
      <input type="text" class="css-input" v-model="search" @input="filter" />
    </section>
    <Event
      
      class="events-section"
      v-for="(event, index) in getEvents"
      :key="index"
      :event="event"
    />
  </section>
</template>

<script>
import Event from "@/components/Event.vue";
export default {
  mounted() {
    this.filter();
  },
  name: "MainPage",
  data() {
    return {
      search: "",
      filterArray: Array,
    };
  },
  components: {
    Event,
  },

  methods: {
   
    filter() {
      let events = this.getEvents;
      if (
        this.search === undefined ||
        this.search === null ||
        this.search == ""
      ) {
        this.filterArray = events;
        console.log("events filter ", this.filterArray);
      } else {
        this.filterArray = events.filter((event) => {
          let titleName = event.name
            .toLowerCase()
            .includes(this.search.toLowerCase);
          console.log("titleName", titleName);
          console.log();
          return titleName;
        });
      }
    },
  },
  computed: {
    getEvents() {
      return this.$store.getters.events;
    },
  },
};
</script>

<style scoped></style>
