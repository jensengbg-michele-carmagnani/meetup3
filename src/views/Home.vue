<template>
  <div id="home">
    <header class="header">
      <h1>Meetup page</h1>
      <nav></nav>
    </header>
    <MainPage />
    <Foot/>
  </div>
</template>

<script>
// @ is an alias to /src
import MainPage from "@/components/MainPage.vue";
import Foot from "@/components/Foot.vue";
export default {
  name: "Home",
  components: {
    MainPage,
    Foot,
  },
 
};
</script>
