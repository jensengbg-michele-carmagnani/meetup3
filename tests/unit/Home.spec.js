import { createLocalVue, mount, shallowMount } from "@vue/test-utils";
import VueRouter from "vue-router";
import Foot from "@/components/Foot.vue";
import MainPage from "@/components/MainPage.vue";
import Home from "@/views/Home.vue";
import App from "@/App.vue";
import index from "@/store/index.js";
import Vuex from "vuex";


const store = new Vuex.Store({ index });
describe("MainPage.vue", () => {
  let localVue;
  
  beforeEach(() => {
    localVue = createLocalVue();
  });
  
  it("should, when the App component is mounted, dispach an action to Vuex", () => {
    localVue.use(VueRouter);
    const router = new VueRouter();
    
    let actions = {
      fetchEvents: jest.fn(),
    };
    const store = new Vuex.Store({ actions });
    const wrapper = shallowMount(App, {
      store,
      localVue,
      router,
    });
    expect(actions.fetchEvents).toHaveBeenCalled();
  });

  
  it("should render Foot coponent", () => {
    const wrapper = mount(Home, {
      store,
      localVue,
    });
    expect(wrapper.html()).toMatchSnapshot();
  });

  it("should render a footer component", () => {
    const wrapper = mount(Home, {
      store,
      localVue,
    });
    const footerComp = wrapper.findComponent(Foot);
    expect(footerComp.exists()).toBe(true);
  });

  it("should render the Event using a real Vuex getter", () => {
    const wrapper = mount(Home, {
      store,
      localVue,
    });
    const eventComp = wrapper.findComponent(MainPage);
    expect(eventComp.exists()).toBe(true);
  });


});
