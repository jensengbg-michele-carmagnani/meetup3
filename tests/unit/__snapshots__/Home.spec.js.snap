// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainPage.vue should render Foot coponent 1`] = `
<div id="home">
  <header class="header">
    <h1>Meetup page</h1>
    <nav></nav>
  </header>
  <section id="main-page">
    <h2 class="title">Eventpage</h2>
    <section class="search-section"><input type="text" class="css-input"></section>
  </section>
  <section id="foot">
    <h1>footer</h1>
    <ul>
      <li></li>
      <li></li>
      <li></li>
      <li></li>
      <li></li>
      <li></li>
    </ul>
  </section>
</div>
`;
